xmtp_mls/src/groups/group_permissions.rs

```diff
@@ -6,11 +6,215 @@
         AndCondition as AndConditionProto, AnyCondition as AnyConditionProto,
         BasePolicy as BasePolicyProto, Kind as PolicyKindProto,
     },
-    MembershipPolicy as MembershipPolicyProto, PolicySet as PolicySetProto,
+    metadata_policy::{
+        AndCondition as MetadataAndConditionProto, AnyCondition as MetadataAnyConditionProto,
+        Kind as MetadataPolicyKindProto, MetadataBasePolicy as MetadataBasePolicyProto,
+    },
+    MembershipPolicy as MembershipPolicyProto, MetadataPolicy as MetadataPolicyProto,
+    PolicySet as PolicySetProto,
 };
 
 use super::validated_commit::{AggregatedMembershipChange, CommitParticipant, ValidatedCommit};
 
+// A trait for policies that can update Metadata for the group
+
+pub trait MetadataPolicy: std::fmt::Debug {
+    fn evaluate(&self, actor: &CommitParticipant, change: bool) -> bool;
```

Comment:
I would recommend having a struct for the change, which includes the new value (and possibly the previous value).

It also seems like if there is no change, we should not even need to evaluate the policy.

The simple boolean works fine when the policy is just "can I change the group name", but it's easy to imagine more complicated policies being required. For example, if the value is an array we may want to evaluate whether items have been added or removed, not just if there is an equality.

Reply:
@neekolas commit history in this PR was off since it was made off a branch that was squash merged to main (I'll try graphite on my next multi PR change)

I cherry-picked changes to branch here => https://github.com/xmtp/libxmtp/pull/677, and I believe I addressed your concerns in this commit https://github.com/xmtp/libxmtp/pull/677/commits/f6ae8d8430cab499a95e3c0438def42c703aa4b2